### PR TITLE
feat: add azure datasource fields

### PR DESCRIPTION
**Pull request by aymericDD**
Add support to create azure datasource. 

Issue : https://github.com/grafana/terraform-provider-grafana/issues/345

### CODE REVIEW

File: datasource.go [hunk 1]
```diff
@@ -35,10 +35,13 @@ type DataSource struct {
 	// Helper to read/write http headers
 	HTTPHeaders map[string]string `json:"-"`
 
-	JSONData       JSONData       `json:"jsonData,omitempty"`
-	SecureJSONData SecureJSONData `json:"secureJsonData,omitempty"`
+	JSONData         JSONData         `json:"jsonData,omitempty"`
+	SecureJSONData   SecureJSONData   `json:"secureJsonData,omitempty"`
+	SecureJSONFields SecureJSONFields `json:"secureJsonFields,omitempty"`
```
Comment (julienduchesne) [inline datasource.go]: `secureJsonFields` is a response field. You do not have to set it on the request and we do not need it in the response (the terraform provider only sets secure fields, it doesn't read them)

Comment (aymericDD) [reply datasource.go]: My bad! :angel:  Thanks for your review :pray:

**Review by julienduchesne — COMMENTED** [review]
Missing clientSecret as a SecureJsonData field, right?

**Comment by aymericDD** [comment]
> Missing clientSecret as a SecureJsonData field, right?

My bad! Fixed :+1:

**Review by julienduchesne — APPROVED** [review]
LGTM!

**Pull request merged**